src/Nelmio/Alice/Instances/Populator/Methods/ArrayAdd.php

```diff
@@ -84,4 +85,37 @@ private function findAdderMethod($object, $property)
             return $method;
         }
     }
+
+    /**
+     * @param string $plural
+     *
+     * @return string
+     */
+    private function singularize($plural)
+    {
+        if (null === $this->singularizeClass || null === $this->singularizeMethod) {
+            // default to false so we only try this once
+            $this->singularizeClass = false;
```

Comment:
Oh I see my bad. Erm in that case this could done in the constructor? If done in `findAdderMethod()` it has the benefit to be done only if required but you have an if check on each usage.